### DISCUSSION TITLE
Topology Json Nullables

### DISCUSSION THREAD

**Discussion by damccull** [ravendb/ravendb#14997] [Q&A]
Is there a list somewhere that describes which fields are allowed to be nullable or omitted and which are always required to exist and have non-null data in the topology information json?

**Comment by garayx** [comment]
Hi

no, we don't document such information

**Comment by ayende** [comment]
What are you trying to do?

**Reply by damccull** [reply]
I'm trying to deserialize the topology Json into some rust structures. It's working great until I run into a case where the server returns null on a field. Then I set that field to optional and try again, but I have run into this happening several times. I'm looking for an understanding of when fields can be null or omitted so I can make them optional in my code without needing to make every single field optional.

**Reply by ayende** [reply]
Aside from `Members`, pretty much everything there can be null, I believe.

**Reply by damccull** [reply]
Ah, ok then. Thank you. I'll go ahead and make them all optional then. I appreciate it!
